src/arch/xtensa/ipc.c

```diff
@@ -0,0 +1,22 @@
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// Copyright(c) 2022 Intel Corporation. All rights reserved.
+//
+// Author: Guennadi Liakhovetski <guennadi.liakhovetski@linux.intel.com>
+
+/**
+ * @file
+ * @brief Xtensa IPC per-core data access
+ */
+
+#include <sof/ipc/common.h>
+#include <sof/lib/alloc.h>
+#include <sof/lib/cpu.h>
+#include <xtos-structs.h>
+
+struct ipc_core_ctx **arch_ipc_get(void)
+{
+	struct core_context *ctx = (struct core_context *)cpu_read_threadptr();
+
+	return &ctx->ipc;
```

Comment:
Why do we need for xtos ?

Reply:
It is currently possible to build IPC4 with XTOS, right? And I think that configuration is or was also used for the initial development / testing? So I thought I shouldn't break that functionality

Reply:
Lets use Zephyr for this work

Reply:
@lgirdwood well, sure, I wouldn't mind that, but what exactly does that mean? Does that mean that it should only be possible to build SOF with IPC4 with Zephyr? So we should disable building IPC4 with XTOS? If we don't do that and keep the possibility to build SOF with IPC4, then how would error handling work there?

Reply:
@lyakh IPC4 with XTOS is not needed.

Reply:
@lgirdwood I just learned from @ujfalusi that work is ongoing for IPC4 with XTOS

Reply:
@lyakh, to clarify: testing is done with both xtos and zephyr, but I personally build SOF with XTOS for my local testing. The parts that I'm working with are way above anything xtos/zephyr and makes no difference for the end result.

Reply:
@ujfalusi ok, got it, but IIUC it might happen that IPC4 will not even build with XTOS beginning from some point. In fact maybe we should add a Kconfig dependency for that.